source/extensions/filters/http/rate_limit_quota/config.cc

```diff
@@ -73,13 +73,23 @@ Http::FilterFactoryCb RateLimitQuotaFilterFactory::createFilterFactoryFromProtoT
         return tl_global_client;
       });
 
-  return [&, config = std::move(config), config_with_hash_key,
-          tls_store](Http::FilterChainFactoryCallbacks& callbacks) -> void {
+  RateLimitOnMatchActionContext action_context;
+  RateLimitQuotaValidationVisitor visitor;
+  Matcher::MatchTreeFactory<Http::HttpMatchingData, RateLimitOnMatchActionContext> matcher_factory(
+      action_context, context.serverFactoryContext(), visitor);
+
+  Matcher::MatchTreeSharedPtr<Http::HttpMatchingData> matcher = nullptr;
+  if (config->has_bucket_matchers()) {
+    matcher = matcher_factory.create(config->bucket_matchers())();
+  }
+
+  return [&, config = std::move(config), config_with_hash_key, tls_store,
+          matcher](Http::FilterChainFactoryCallbacks& callbacks) -> void {
```

Comment:
for the `matcher` shared pointer here, you only need to make copy (i.e., increase ref count once).

For example, in the RateLimitQuotaFilter constructor, it is passed by value which will achieve the need of copy above. Other places can just be move